### ISSUE TITLE
Indicator setup: UI improvements to disaggregation selection menu

### ISSUE THREAD

**Issue by jennyhavah** [mercycorps/TolaActivity#1213]
**Note:** This ticket is dependent on #1849 being completed.

# Stories

1. I can review and opt out of global disaggregations when they do not make sense for a particular indicator. (Right now, global disaggs are not even displayed in the disaggregations menu, and are automatically selected for every indicator, regardless of relevance.)

2. While browsing disaggregation options, I can easily see what the categories are for any given disaggregation type.

**Affects:** Indicator setup form > Performance tab > Disaggregation multi-select. Here is how it looks today -- a flat list of all Custom disaggregations. Global disaggregations such as SADD are not displayed.

![image](https://user-images.githubusercontent.com/33670923/53065774-f194d800-3481-11e9-9c6d-30b604ab69ed.png)

# Acceptance criteria

The proposed design replaces the multi-select with a simple list of checkboxes, showing all disaggregations available to this indicator:

![image](https://user-images.githubusercontent.com/33670923/61654240-d7e5ff80-ac70-11e9-870d-4b3f9edb09eb.png)

# Acceptance criteria

- [x] 1. Global disaggregations come first, followed by Country disaggregations.

     - [x] 1.1. All global disaggregations are _checked by default_, but may be unchecked.

     - [x] 1.2. Whenever additional global disaggregation types are introduced, they should be listed in alphabetical order. (To test, we'll need to add a fake second global disagg type in Django admin on the QA server.)

     - [x] 1.3. We should have only one global disaggregations -- SADD -- that should be fully spelled out as illustrated in the mockup. 

          - [x] 1.3.1 There should not be a "---" or a "By Municipality" disaggregation. (If there are any results linked to the latter disaggregation options, we should make that a country-level disaggregation and reassign results accordingly.)

          - [x] 1.3.2. All sections should be OPEN by default, with the caret pointed downwards. (Carets should always point downward when the item is open.) I neglected to specify that -- sorry. Unlike the results form, where they're closed by default, here they should be open. (We're limiting the vertical height of the field, so we don't need to worry about it stretching out the form to a crazy length... And we don't want this option to be easily missed, which could happen if it's closed by default.)

          - [x] 1.3.3. Font size for disaggregation options is smaller than font size used for form labels and disaggregation collapso-expandos.

- [x] 3. Country disaggregation groupings are displayed as "[CountryName] disaggregations". Note: Because the vast majority of programs are based in a single country, I see no need to over-complicate the UI with nested collapso-expandos. As shown in the mockup, it's fine to display in a single list: Global, Country 1, Country 2  

     - [x] 3.1. When there are multiple country disaggregations, the countries are listed in alphabetical order. To emphasize, Global always comes first, followed by the alphabetized list of country disaggs.  <P>

- [x] 4. If there are a more than [n] available disaggregations *(what is [n]?)*, they could be grouped in a scrolling div to prevent the form from growing very long.

- [x] 5. There should be no "Select all" option. More is not better in this case, so we shouldn't make it easy to select all of them. (M&E advisors often have to restrain program teams from getting over-ambitious with disaggregated data, committing to more granular data collection than their M&E budget can support.)

- [x] 6. A “Preview” of the labels (options) for a disaggregation appear in a tooltip when the user clicks on an info icon.

     ![image](https://user-images.githubusercontent.com/169060/53987847-fb037e80-40d6-11e9-804a-4a96493304b3.png)

- [x] 7. When the following criteria are met, the checkbox is **_checked but inactive_**: There is at least one result submitted for _that particular indicator_ that includes at least one disaggregated value from _that specific disaggregation_. 

     _Right now, best I can tell, unchecking a disaggregation box has no effect on saved results, so we're doing the right thing on the back end in not permitting you to delete data by unchecking a disaggregation box. It's still problematic to allow that deselection after results are added, because you can easily end up with a situation where results reported _for the same indicator_ have varied categories -- with the indicator form not accurately reflecting what's available. It also seems problematic for IPTT queries that need to figure out which categories to display on a _per indicator_ basis, not a _per result_ basis._

     - [x] 7.1. When a disaggregation is inactive, include this sentence after the preview list: This disaggregation cannot be unselected, because it was already used in submitted program results.

     ![image](https://user-images.githubusercontent.com/33670923/61655011-8b032880-ac72-11e9-90ed-cc758b7ab8e0.png)

     - [x] 7.2. In this scenario, if I go ahead and deselect a disaggregation for an indicator, it no longer appears in any result forms -- new or saved. (Right now, it doesn't appear in new results for the indicator, but continues to appear in saved results.)

     - [x] 7.3. Conversely, if I _add_ a new disaggregation to an indicator, it should appear in both new and saved results. (Right now, this is the case UNLESS a disaggregated value of any kind has been entered under Custom disaggregations. So if a program forgets to check a disaggregation and starts reporting results, there is no way for them to update existing results later on, with disaggregated data. Note: I filed a separate bug covering this: #2054. It's possible that was overkill and it's a moot point with the way we're restructuring everything)

## Random UI thing while you're in the area

- [x] 1. Please update the label to sentence case for "Rationale or justification for indicator", immediately above the disaggregation section. (It's currently title case on Prod.)

## Housekeeping
* [ ]  Unit test/s written wherever applicable
* [ ]  i18n tags included wherever applicable
* [ ]  Functionality is accessible to people using screen readers

**Comment by sanjuroj** [comment]
Either we convert the indicator modal to React, which makes this ticket easier (but still not trivial). Or we leave it as non-React, in which case this is even more work.  Either way a high level of effort.

**Comment by jennyhavah** [comment]
@sanjuroj Assuming you closed the ticket by accident. :-)

**Comment by sanjuroj** [comment]
Gah, yes, sorry. Second time I've done that.

**Comment by camgnostic** [comment]
Moving to acceptance testing despite one unclear spec (number 4) @jennyhavah  - wanna field test and see if you want to pick an "n"?

**Comment by jennyhavah** [comment]
@camgnostic I added a section 1.3 to address some issues I just noticed just now. The item about Global having some unwanted disagg types was based on seeing this.

![image](https://user-images.githubusercontent.com/33670923/71372486-5f5dd480-2569-11ea-81b0-bd206a6e5b58.png)

I also noticed that the caret is not responding when I open sections -- also captured in 1.3.

**Comment by jennyhavah** [comment]
> Moving to acceptance testing despite one unclear spec (number 4) @jennyhavah - wanna field test and see if you want to pick an "n"?

I can't see a list with lots of options at the moment. For starters, let's go with what's in the mockup -- 8 options visible, any more are enclosed in the enclosing box thingy.

Please note that the font size for the options are smaller than the form field labels. (A little hard to see in the mockup, maybe.)

**Comment by jennyhavah** [comment]
Requested UI refinement:

Please remove white fill you currently see inside the selection box when interacting with the collapso-expando. Here's how it looks when I click it open:

![image](https://user-images.githubusercontent.com/33670923/71423143-03b04b80-263c-11ea-93fc-56cebfb79112.png)

And when I close it again:

![image](https://user-images.githubusercontent.com/33670923/71423241-d1531e00-263c-11ea-821c-eb56f1b60cd9.png)

For an example of how I'd like it to look, check out the lack of fill on the program page, like so:

![image](https://user-images.githubusercontent.com/33670923/71423218-a4067000-263c-11ea-99d8-45ff24029e25.png)

I'd actually prefer there not be a blue selection box either, but only if it doesn't require weird hax or workarounds to accomplish. (Not sure if Paul already attempted this or not.)

**Comment by jennyhavah** [comment]
Reminder to self: Retest "Can I still view selection as a Low/Med" user after refinements are done.

**Comment by jennyhavah** [comment]
Some additional UI refinements:

![image](https://user-images.githubusercontent.com/33670923/71496354-c7c9e180-2807-11ea-8d6c-7ebb316a99ad.png)

1. Add space between global and country level disaggregation "clusters".

2. Add space between Disaggregation and Type fields.

3. Add border and scrollbar to disaggregation list, so it's more obvious that it's a scrollable list. The original mockup has the thing partially obscured by the tooltip. Here's a mockup without the tooltip so you can see everything more clearly:

![image](https://user-images.githubusercontent.com/33670923/71496434-22fbd400-2808-11ea-84b9-74136cce6e87.png)

**Comment by camgnostic** [comment]
Passing to @axoplasm for the last round of UI tweaks

**Comment by jennyhavah** [comment]
@axoplasm I created a new ticket just for the remaining UI tweaks so none of us have to look at this ticket everrrr again. See #2091.

Moving this one to Passed.
